gammapy/irf/energy_dispersion.py

```diff
@@ -8,170 +8,96 @@
-            ' can lead to incorrect results.\n'
-            ss += 'Old PDF threshold: {0}\n'.format(self._pdf_threshold)
-            ss += 'New PDF threshold: {0}'.format(value)
-            raise Exception(ss)
-
-        self._pdf_threshold = value
-
-        m = self._pdf_matrix
-        m[m < value] = 0
-
-    @property
-    def reco_energy(self):
-        """Reconstructed Energy axis (`~gammapy.utils.energy.EnergyBounds`)
-        """
-        return self._e_reco
-
-    @property
-    def true_energy(self):
-        """Reconstructed Energy axis (`~gammapy.utils.energy.EnergyBounds`)
-        """
-        return self._e_true
-
-    def __str__(self):
-        ss = 'Energy dispersion information:\n'
-        ss += 'PDF matrix threshold: {0}\n'.format(self.pdf_threshold)
-        m = self._pdf_matrix
-        ss += 'PDF matrix filling factor: {0}\n'.format(np.sum(m > 0) / m.size)
-        ss += 'True energy min: {:.2f}\n'.format(self.reco_energy[0])
-        ss += 'Reco energy min: {:.2f}\n'.format(self.true_energy[0])
-        return ss
+        return self.data
 
     @classmethod
-    def from_gauss(cls, ebounds, sigma=0.2):
-        """Create gaussian `EnergyDispersion` matrix.
+    def from_gauss(cls, e_true, e_reco, sigma=0.2, pdf_threshold=1e-6):
+        """Create Gaussian `EnergyDispersion` matrix.
 
-        TODO: Debug
-        TODO: this is Gaussian in e_reco ... should be log(e_reco) I think.
+        The output matrix will be Gaussian in log(e_true / e_reco)
+
+        TODO: extend to have a vector of bias various true energies.
+        TODO: extend to have vector of  resolution for various true energies.
         TODO: give formula: Gaussian in log(e_reco)
         TODO: add option to add poisson noise
-        TODO: extend to have a vector of bias and resolution for various true energies.
 
         Parameters
         ----------
-        e_reco : `~gammapy.utils.energy.EnergyBounds`
-            Reco and true energy binning
-        sigma : float
-            RMS width of Gaussian energy dispersion.
+        e_true : `~astropy.units.Quantity`, `~gammapy.utils.nddata.BinnedDataAxis`
+            Bin edges of true energy axis
+        e_reco : `~astropy.units.Quantity`, `~gammapy.utils.nddata.BinnedDataAxis`
+            Bin edges of reconstructed energy axis
+        sigma : float, optional
+            RMS width of Gaussian energy dispersion, resolution
+        pdf_threshold : float, optional
+            Zero suppression threshold
         """
         from scipy.special import erf
 
-        e_bounds = ebounds.value
+        # Init array without data
+        retval = cls(e_true = e_true, e_reco = e_reco)
 
-        nbins = len(e_bounds) - 1
-        logerange = np.log10(e_bounds)
+        # erf does not work with Quantities
+        reco = retval.e_reco.data.to('TeV').value
+        true = retval.e_true.nodes.to('TeV').value
+        migra_min = np.log10(reco[:-1] / true[:,np.newaxis])
```

Comment:
@registerrier 
Can you please have a look at this? I implemented the ``from_gauss`` method so that 
```
log10(E_true / E_reco)
```

follows a Gaussian distribution. Is this correct? The picture looks like this for

```
e_true = np.logspace(0,1, 101) * u.TeV 

edisp = EnergyDispersion.from_gauss(e_true=e_true, e_reco=e_true, sigma=0.1)
```

![example](https://cloud.githubusercontent.com/assets/11678435/16695307/563be9b8-453f-11e6-85cc-c2810583c940.png)